pkg/gadgettracermanager/gadgettracermanager.go

```diff
@@ -0,0 +1,169 @@
+package gadgettracermanager
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"sync"
+	"unsafe"
+
+	bpflib "github.com/iovisor/gobpf/elf"
+	_ "github.com/iovisor/gobpf/pkg/bpffs"
+	_ "github.com/iovisor/gobpf/pkg/cpuonline"
+
+	pb "github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/api"
+)
+
+type GadgetTracerManager struct {
+	mu sync.Mutex
+
+	// containers by cgroupId
+	containers map[uint64]pb.ContainerDefinition
+
+	// tracers by filterId
+	tracers map[string]tracer
+}
+
+type tracer struct {
+	filterId string
+
+	containerSelector pb.ContainerSelector
+
+	mapHolder          *bpflib.Module
+	cgroupIdSetMap     *bpflib.Map
+	cgroupIdSetMapPath string
+
+	matchesCache []uint64
+}
+
+func containerSelectorMatches(s *pb.ContainerSelector, c *pb.ContainerDefinition) bool {
```

Comment:
TODO: check if we can use upstream kubernetes packages for this:

https://godoc.org/k8s.io/apimachinery/pkg/labels#ParseToRequirements
https://godoc.org/k8s.io/apimachinery/pkg/labels#Set
https://godoc.org/k8s.io/apimachinery/pkg/labels#Selector